tests/integration/ha/test_ha.py

```diff
@@ -249,6 +252,61 @@ async def test_freeze_broker_with_topic_leader(
     assert_continuous_writes_consistency(result=result)
 
 
+@pytest.mark.abort_on_fail
+async def test_network_cut_without_ip_change(
+    ops_test: OpsTest,
+    c_writes: ContinuousWrites,
+    c_writes_runner: ContinuousWrites,
+):
+    # Let some time pass for messages to be produced
+    await asyncio.sleep(10)
+
+    topic_description = await get_topic_description(
+        ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME
+    )
+    initial_leader_num = topic_description.leader
+    leader_machine_name = await get_unit_machine_name(
+        ops_test=ops_test, unit_name=f"{APP_NAME}/{initial_leader_num}"
+    )
+
+    logger.info(
+        f"Throttling network for leader of topic '{ContinuousWrites.TOPIC_NAME}': {initial_leader_num}"
+    )
+    network_throttle(machine_name=leader_machine_name)
+    await asyncio.sleep(REELECTION_TIME * 2)
+
+    # verify replica is not in sync
+    topic_description = await get_topic_description(
+        ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME
+    )
+    assert topic_description.in_sync_replicas == {0, 1, 2} - {initial_leader_num}
+    assert initial_leader_num != topic_description.leader
```

Comment:
**nitpick/suggestion:** Tiny suggestion, maybe add an assertion failed message here saying something like 'topic leader not re-elected' or something to communicate that.

Reply:
Yes, I think I need to do a pass on the whole file to cleanup and improve error messages.

Reply:
Nice. I don't think they're **always** necessary, but they're useful when we're doing something abstract like comparing two `int`s, and using that information to decide about something more complex (like leader election)